src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/RazorDefinitionEndpoint.cs

```diff
@@ -128,12 +131,81 @@ public DefinitionRegistrationOptions GetRegistrationOptions(DefinitionCapability
                 new LocationOrLocationLink(new Location
                 {
                     Uri = originComponentUri,
-                    Range = new Range(new Position(0, 0), new Position(0, 0)),
+                    Range = range,
                 }),
             });
         }
 
-        internal static async Task<TagHelperBinding?> GetOriginTagHelperBindingAsync(
+        private async Task<Range> GetNavigateRangeAsync(DocumentSnapshot documentSnapshot, BoundAttributeDescriptor? attributeDescriptor, CancellationToken cancellationToken)
+        {
+            if (attributeDescriptor is not null)
+            {
+                var originCodeDocument = await documentSnapshot.GetGeneratedOutputAsync().ConfigureAwait(false);
+                var range = await TryGetPropertyRangeAsync(originCodeDocument, attributeDescriptor.GetPropertyName(), _documentMappingService, cancellationToken).ConfigureAwait(false);
+
+                if (range is not null)
+                {
+                    return range;
+                }
+            }
+
+            return new Range(new Position(0, 0), new Position(0, 0));
```

Comment:
At first I was against this default but I think the plan is "If we can't figure out exactly where you should end up lets at least put you at the top of the correct .razor file"? If so I think that's an excellent plan, but a little comment here to explain that that's what's happening would be good.

Reply:
This is also where you end up if you go to def on the tag name, not an attribute, so it is maintaining current behavior.

But yeah, if you do have an attribute, and my fancy code deosnt find it, we could just bail out completely and not go anywhere, which is what happens now, but it seems a bit silly IMO.

Plus I'm hoping to follow up with support for partial classes etc soon enough. I don't _think_ it will be too hard... maybe.